### DISCUSSION TITLE
Telegram message duplicated in Mattermost

### DISCUSSION THREAD

**Discussion by nikkilocke** [42wim/matterbridge#1430] [Help / Q&A]
I am doing a one-way transfer of messages from a Telegram broadcast channel into Mattermost (so people can see them there, without having to join Telegram).

This message arrived on Telegram:
![image](https://user-images.githubusercontent.com/684025/111827796-3e0da480-88e2-11eb-833d-7128d4e8eb83.png)

It appeared on Mattermost as two identical messages:
![image](https://user-images.githubusercontent.com/684025/111827875-57165580-88e2-11eb-8d20-fe19270d7822.png)

A previous message, with no quoted text, transferred fine as a single message.

Is that a bug, or could it be misconfiguration?

I have triple checked that I haven't got two gateways in my config file.

**Comment by 42wim** [comment]
Is this something you can reproduce?
If so please run with matterbridge `--debug` so we can see what happens in the log

**Reply by nikkilocke** [reply]
I have failed to reproduce the duplicate message (perhaps I stopped Matterbridge in the instant it had posted the message, but before it marked it as skipped?).

However, I have two related issues:
1) The Nickname always comes out as Unknown.
2) The quoted part of the Telegram message is not copied. (But I now wonder if it is not quoted, but that Telegram saw the uri in the message, and may have reproduced some part of the contents of the linked page. In which case, you wouldn't expect it to come across.)

I enclose a debug log of transfer of a copy of that original message posted into a test channel.
[debug.log](https://github.com/42wim/matterbridge/files/6212752/debug.log)
